### PR TITLE
ci: adds workflow to cleanup caches in PRs.

### PR DESCRIPTION
**Pull request by mathetake**

### CODE REVIEW

File: .github/workflows/clear_cache.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#force-deleting-cache-entries
+name: cleanup caches by a branch
+on:
+  # Only runs on the PRs are closed, regardless of it is merged or not.
+  # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-pull_request-workflow-when-a-pull-request-merges
+  pull_request:
+    types:
+      - closed
+
+jobs:
```
Comment (mathetake) [inline .github/workflows/clear_cache.yaml:11]: this is exactly the same 🍝 https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#force-deleting-cache-entries

**Review by codefromthecrypt — APPROVED** [review]
great. you mind clearing the ad-hoc already closed cache entries?

**Comment by mathetake** [comment]
on it

**Comment by mathetake** [comment]
done

**Pull request merged**